Simplenote/Classes/SPNoteEditorViewController+Extensions.swift

```diff
@@ -224,6 +226,62 @@ private extension SPNoteEditorViewController {
 
         SPTracker.trackEditorActivitiesAccessed()
     }
+
+    func presentShareController(for note: Note, from sourceView: UIView) {
+        guard let activityController = UIActivityViewController(note: note) else {
+            return
+        }
+
+        activityController.configureAsPopover(sourceView: sourceView)
+
+        present(activityController, animated: true, completion: nil)
+        SPTracker.trackEditorNoteContentShared()
+    }
+}
+
+
+// MARK: - Services
+//
+extension SPNoteEditorViewController {
+
+    func delete(note: Note) {
+        SPTracker.trackEditorNoteDeleted()
+        SPObjectManager.shared().trashNote(note)
+        CSSearchableIndex.default().deleteSearchableNote(note)
+    }
+}
+
+
+// MARK: - OptionsControllerDelegate
+//
+extension SPNoteEditorViewController: OptionsControllerDelegate {
+
+    func optionsControllerDidPressShare(_ sender: OptionsViewController) {
+        sender.dismiss(animated: true, completion: nil)
+
+        // Wait a bit until the Dismiss Animation concludes. `dismiss(:completion)` takes too long!
+        DispatchQueue.main.asyncAfter(deadline: .now() + UIKitConstants.animationDelayShort) {
+            self.presentShareController(for: sender.note, from: self.actionButton)
+        }
+    }
+
+    func optionsControllerDidPressHistory(_ sender: OptionsViewController) {
+        sender.dismiss(animated: true, completion: nil)
+
+        // Wait a bit until the Dismiss Animation concludes. `dismiss(:completion)` takes too long!
+        DispatchQueue.main.asyncAfter(deadline: .now() + UIKitConstants.animationDelayShort) {
+            self.presentHistoryController()
+        }
+    }
+
+    func optionsControllerDidPressTrash(_ sender: OptionsViewController) {
+        sender.dismiss(animated: true, completion: nil)
+
+        // Wait a bit until the Dismiss Animation concludes. `dismiss(:completion)` takes too long!
+        DispatchQueue.main.asyncAfter(deadline: .now() + UIKitConstants.animationDelayShort) {
+            self.delete(note: sender.note)
```

Comment:
When trashing a note we want to dismiss the editor, yeah?  Are we missing a call to `self.backButtonAction(sender)` here?